presto-main/src/main/java/com/facebook/presto/sql/planner/CanonicalPartitioningScheme.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.connector.ConnectorPartitioningHandle;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.facebook.presto.sql.planner.RowExpressionVariableInliner.inlineVariables;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class CanonicalPartitioningScheme
```

Comment:
We have a lot of `CanonicalXXX` counter part in this commit. Shall we add some tests to make sure the canonical part is in sync with the original class? Meaning that if the original class changes, we should fail some tests or automatically fix the `CanonicalXXX`

Reply:
Great idea. Let me try to add some tests for this.

Reply:
Added some field assertion tests in `TestCanonicalPlanGenerator`